packages/pubsub/src/SchemaDeployer.ts

```
@@ -0,0 +1,115 @@
+      await this.createRevisions(forNewRevision)
+    }
+
+    this.logger.info('Schemas deployment is finished')
+
+  }
+
+  private async createRevisions(forNewRevision: Map<string, string>) {
+    const projectName = process.env['GCLOUD_PROJECT']
+    if (!projectName) {
+      throw new Error('Can\'t find GCLOUD_PROJECT env variable, please define it')
+    }
+    for (const [topicSchema, definition] of forNewRevision) {
+      const schemaName = topicSchema + SCHEMA_NAME_SUFFIX
+      const schemaPath = `projects/${projectName}/schemas/${schemaName}`
+      await this.schemaClient.commitSchema({
+        name: schemaPath, schema: {
+          name: schemaPath, type: 'AVRO', definition,
+        },
+      })
+      this.logger.info(`Schema ${schemaName} is updated`)
+    }
+  }
+
+  private async createSchemas(forCreate: Map<string, string>) {
+    for (const [topicSchema, definition] of forCreate) {
+      const schemaName = topicSchema + SCHEMA_NAME_SUFFIX
+      await this.pubSubClient.createSchema(schemaName, 'AVRO', definition)
+      this.logger.info(`Schema ${schemaName} is created`)
+    }
+  }
+
+  private getEnabledTopicSchemas(schemasConfig: Record<string, boolean>, readerSchemas: Record<string, ReaderAvroSchema>)
+    : Map<string, string> {
+    const enableTopicSchemas = new Map<string, string>()
+    for (const topicName in schemasConfig) {
+      const readerSchema = readerSchemas[topicName]
+      if (schemasConfig[topicName] && readerSchema) {
+        enableTopicSchemas.set(topicName, JSON.stringify(readerSchema.reader))
+      }
+    }
+    return enableTopicSchemas;
+  }
+
+  private async aggregateTopicSchemas(topicSchemasToDeploy: Map<string, string>)
+    : Promise<ISchemasForDeployment> {
+    const forCreate = new Map<string, string>(topicSchemasToDeploy)
+    const forNewRevision = new Map<string, string>()
+    for await (const schema of this.pubSubClient.listSchemas('FULL')) {
+      if (schema.type != 'AVRO') {
```

Maybe we should throw an error if a non-avro schema with the name of one of our event is registered, and throw an error?